app/controllers/mixins/actions/vm_actions/reconfigure.rb

```diff
@@ -181,6 +187,21 @@ def reconfigure_calculations(mbsize)
           return humansize.to_s, fmt
         end
 
+        def get_vlan_options(host_id)
+          vlan_options = []
+
+          # determine available switches for this host...
+          switch_ids = []
+          HostSwitch.where("host_id = ?", host_id).each do |host_switch|
```

Comment:
Missing RBAC? (And with Lan.where)

Reply:
@himdel Could you elaborate on that?

Reply:
@himdel Do you mean add filter_by_tags as in miq_provision_virt_workflow.rb ?

Reply:
Sure, sorry.. I think you need `Rbac.filtered( HostSwitch.where("host_id = ?", host_id) ).each` to filter this only to the switches the user is allowed to see..